tests/unit/test_database.py

```diff
@@ -1792,6 +1791,46 @@ class Testing(Exception):
 
         self.assertIs(pool._session, session)
 
+    def test_context_mgr_session_not_found_error(self):
```

Comment:
Maybe add another test which reflects the case where the `NOT_FOUND` error is unrelated to the session?

Reply:
Added. Thanks :)